pymc_marketing/mmm/base.py

```diff
@@ -231,48 +242,86 @@ def plot_posterior_predictive(
+            if contracted_dims
+            else channel_contributions
+        )
+        means = [channel_contributions.mean(["chain", "draw"])]
+        contribution_vars = [
+            az.hdi(channel_contributions, hdi_prob=0.94).channel_contributions
+        ]
 
         if self.control_columns:
-            contribution_vars.append("control_contribution")
+            control_contributions = az.extract(
+                self.fit_result,
+                var_names=["control_contributions"],
+                combined=False,
+            )
+            contracted_dims = [
+                d
+                for d in control_contributions.dims
+                if d not in ["chain", "draw", "date"]
+            ]
+            control_contributions = (
+                control_contributions.sum(contracted_dims)
+                if contracted_dims
+                else control_contributions
+            )
+            means.append(control_contributions.mean(["chain", "draw"]))
+            contribution_vars.append(
+                az.hdi(control_contributions, hdi_prob=0.94).control_contributions
+            )
 
         fig, ax = plt.subplots(**plt_kwargs)
 
-        for i, var_contribution in enumerate(contribution_vars):
+        for i, (mean, hdi, var_contribution) in enumerate(
+            zip(
+                means,
+                contribution_vars,
+                ["channel_contribution", "control_contribution"],
+            )
+        ):
             ax.fill_between(
                 x=self.data_df[self.date_column],
-                y1=model_hdi[var_contribution][:, 0],
-                y2=model_hdi[var_contribution][:, 1],
+                y1=hdi.isel(hdi=0),
+                y2=hdi.isel(hdi=1),
                 color=f"C{i}",
                 alpha=0.25,
                 label=f"$94 %$ HDI ({var_contribution})",
             )
             sns.lineplot(
                 x=self.data_df[self.date_column],
-                y=az.extract(self.fit_result, var_names=[var_contribution]).mean(
-                    axis=1
-                ),
+                y=mean,
                 color=f"C{i}",
                 ax=ax,
             )
 
-        intercept_hdi: npt.NDArray[np.float_] = np.repeat(
-            a=model_hdi["intercept"].to_numpy()[None, ...],
+        intercept = az.extract(self.fit_result, var_names=["intercept"], combined=False)
+        intercept_hdi = np.repeat(
+            a=az.hdi(intercept).intercept.data[None, ...],
             repeats=self.n_obs,
             axis=0,
         )
         sns.lineplot(
             x=self.data_df[self.date_column],
-            y=az.extract(self.fit_result, var_names=["intercept"]).mean().item(),
+            y=intercept.mean().data,
             color=f"C{i + 1}",
             ax=ax,
         )
         ax.fill_between(
             x=self.data_df[self.date_column],
             y1=intercept_hdi[:, 0],
-            y2=intercept_hdi[:, 1],
+            y2=intercept_hdi[:, 0],
```

Comment:
seems `y1` is the same as `y2` now? maybe this is not intentional?

Reply:
Oops, that was a typo